src/components/fieldset/fieldset.ts

```diff
@@ -0,0 +1,35 @@
+import { Input, OnInit } from '@angular/core';
+import { BaseComponent, BaseElement, BaseOptions } from '../base';
+import { FormioComponents } from '../components';
+import { FormioTemplate } from '../../formio.template';
+import { FormArray, FormGroup, FormControl } from '@angular/forms';
+
+export interface FieldSetOptions extends BaseOptions<any> {
+    components: Array<any>,
+    legend?: string
+}
+
+export class FieldSetComponent extends BaseComponent<FieldSetOptions> {
+    getControl(): FormArray | FormGroup | FormControl {
```

Comment:
You will now need the following so that the rendering event occurs.

```
get numComponents() : number {
    return this.component.settings.components.length;
}
```